engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EventRepository.java

```diff
@@ -17,4 +55,103 @@ public Event create(Event event) {
         statelessSessionFactory.getCurrentSession().insert(event);
         return event;
     }
+
+    /**
+     * Finds the events related to the provided org id in order to export them.
+     * It is the caller's responsibility to provide valid "from" and "to"
+     * filters.
+     * @param orgId the org id the events are related to.
+     * @param from the initial date to filter the dates from.
+     * @param to the final date to filter the dates from.
+     * @return a list of events that comply with the provided filters.
+     */
+    public List<Event> findEventsToExport(final String orgId, final LocalDate from, final LocalDate to) {
+        this.validateFromTo(from, to);
+
+        final CriteriaBuilder criteriaBuilder = this.session.getCriteriaBuilder();
```

Comment:
I'm not against using criterias but this means adding a 4th way of writing Hibernate queries (in addition to native SQL, HQL and the `QueryBuilder` API @josejulio added to Notifications a while ago). Is that really necessary?

Reply:
FWIW the tests of the stuff I added a while ago: https://github.com/RedHatInsights/notifications-backend/blob/master/backend/src/test/java/com/redhat/cloud/notifications/db/builder/QueryBuilderTest.java

Reply:
I have pushed the refactored code using HQL. I simply did it using criteria because it felt cleaner than using parameter maps and appending strings together for the query, no other special reason.